src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloud.java

```diff
@@ -164,8 +164,6 @@ protected Object readResolve() {
 
                 // Apply a label that identifies the name of this instance configuration
                 configuration.appendLabel(CONFIG_LABEL_KEY, configuration.getNamePrefix());
-                configuration.appendLabel(
-                        CleanLostNodesWork.NODE_IN_USE_LABEL_KEY, CleanLostNodesWork.getLastRefreshLabelVal());
```

Comment:
this was the bug, this is not run every time vm is provisioned

Reply:
Note to reviewers: https://github.com/jenkinsci/google-compute-engine-plugin/blob/9da02e4044e45062d699f24fcf9c8a360bde5f12/src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloud.java#L231-L235 also calls this method for freshly saved configuration, which is unusual. (Normally `readResolve` is limited to fixups that might need to be made when reading an existing configuration from disk, and if both cases need to share some common code, it is factored into a method without a magic `@Serial` name, like `init()` or similar.) Therefore prior to fix, this one call site of `getLastRefreshLabelVal` covered both fresh and loaded configuration; now the one call site is in `InstanceConfiguration.instance`, when a specific VM is provisioned.